mantidproject/mantid
# Add undo fit button to Muon Analysis

**Description of work.**

This adds a undo fit button to the Muon Analysis GUI. It also refactors some of the internal tracking of fits to enable this change.

It also removes the global column when doing single and sequential fits. 

It also removes the clearing of fits when switching between sequential and single mode.



**To test:**

* You will need Access to the ISIS data archive to test this PR.
  * Open the Muon Analysis 2 interface and load some data (MUSR62260 tends to be a good example)
  * Go to the fitting tab add the GaussOsc function to the function browser (it's in the Muon subsection)
  * Click Fit
  * Check that the fitted line and the Diff appear on the graph. 
  * Click undo fit and check that all the values are reset to what they were before the latest fit and that the fitting curve is removed from the graph.
  * Type in 62260-2 in the run selection box
  * Do a sequential fit, then switch to simultaneous mode, check that all the parameters in the function browser remain the same but that the globals column appears.



Fixes #26031  


*This does not require release notes* because this adds functionality to a new GUI.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
